src/KustoExpressionParser.ts

```diff
@@ -43,17 +43,30 @@ export class KustoExpressionParser {
     };
 
     const parts: string[] = [];
+    const expandParts: string[] = [];
+    const isDynamicArray = query.search.property.name.includes(DYNAMIC_TYPE_ARRAY_DELIMITER);
+    const name: string = this.addExpanPartsdIfNeeded(query.search.property.name, expandParts);
+    const column = context.castIfDynamic(name, query.search.property.name);
     this.appendProperty(context, query.expression.from, parts);
     this.appendTimeFilter(context, undefined, parts, tableSchema);
+    this.appendMvExpand(expandParts, parts);
+
+    if (expandParts.length) {
+      // Replace the column name in the search expression with the "expanded" column name
+      query.search.property.name = name;
+    }
 
-    if (query.index) {
+    //query.index is used by the legacy query editor
+    //isDynamicArray keeps the old behavior of the legacy query editor for non dynamic arrays
```

Comment:
I don't really understand this comment? I guess that for arrays, you are not attempting to replace the unfinished `where` condition (the one we are trying to get suggestions for) from the list of filters, but I am not sure why?

Reply:
Yeah sorry it wasn't clear. I updated the comment so hopefully it's more clear? Essentially we're checking for the case of: 

1. the legacy query editor
2. a non-dynamic value, e.g. "ActivityName" 

If those two conditions are met we need to replace by index for autocomplete to continue to work correctly. Here's a screenshot of the case I'm talking about to hopefully clear up what I mean (words are hard today :see_no_evil: ).
![image](https://user-images.githubusercontent.com/1048831/193832838-f608c324-c3bd-4eac-8611-0b8a30603458.png)

Reply:
yeah, my question is, why is it different for dynamic arrays? In your example, if you say `ActivityName == Football` and add a new filter `Modes[] == `, then as suggestions you will get all the possible values, including for rows in which `ActivityName` is not `Footbal` (which is the case anyway for the new editor because we are not passing down that index yet). It's not a blocking thing, but just curious why it was that way (in case we add the index in the new editor).

Reply:
cool! Thanks for checking!